pkg/broker/ingress/ingress_handler.go

```diff
@@ -223,16 +273,15 @@ func (h *Handler) receive(ctx context.Context, headers http.Header, event *cloud
 
 	channelAddress, err := h.getChannelAddress(brokerName, brokerNamespace)
 	if err != nil {
-		h.Logger.Warn("Failed to get channel address, falling back on guess", zap.Error(err))
-		channelAddress = guessChannelAddress(brokerName, brokerNamespace, network.GetClusterDomainName())
+		h.Logger.Warn("Broker not found in the namespace", zap.Error(err))
 	}
 
-	return h.send(ctx, headers, event, channelAddress)
+	return h.send(ctx, headers, event, *channelAddress)
```

Comment:
@Vishal-Chdhry could you take a look at the failing unit test?

>   === FAIL: pkg/broker/ingress TestHandler_ServeHTTP/no_broker_annotations (0.00s)
=== FAIL: pkg/broker/ingress TestHandler_ServeHTTP (0.08s) panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	
> panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x273574b]
goroutine 77 [running]:
testing.tRunner.func1.2({0x2959a80, 0x3f81f90})
	/root/.gvm/gos/go1.20.4/src/testing/testing.go:1526 +0x372
testing.tRunner.func1()
	/root/.gvm/gos/go1.20.4/src/testing/testing.go:1529 +0x650
panic({0x2959a80, 0x3f81f90})
	/root/.gvm/gos/go1.20.4/src/runtime/panic.go:890 +0x263
knative.dev/eventing/pkg/broker/ingress.(*Handler).receive(0xc00058e8c0, {0x2fdf750, 0xc0006adb90}, 0xc000228e82?, 0xc00058e900, {0xc000154876, 0x2}, {0xc000154879, 0x4})
	/home/prow/go/src/knative.dev/eventing/pkg/broker/ingress/ingress_handler.go:279 +0x8eb 

It seems like the problem is on this line, and is probably from deferencing a `nil` channelAddress when there are no broker annotations. WDYT?

Reply:
I think this is because of the fact that we are [not guessing the channel address anymore](https://github.com/knative/eventing/pull/6986#pullrequestreview-1486391056). I have updated the function to return an status code 400 with noDuration in this case: [`0faea53` (#6986)](https://github.com/knative/eventing/pull/6986/commits/0faea53206f247ab9629b6a2ab3ac93af5ba5a09)